### PR TITLE
upgrade laterals and boundary conditions

### PR DESCRIPTION
**Pull request by elisalle**

### CODE REVIEW

**Review by margrietpalm — CHANGES_REQUESTED** [review]
The migration and schema changes don't fully match the specifications and there is an issue with changing names. Also there are no changes added to `CHANGES.rst`.

File: threedi_schema/migrations/versions/0225_migrate_lateral_boundary_condition_tables.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+
+# old name, new name
+# the columns will be renamed using raw sql
+# this is because alembic has conniptions whenever you try to batch rename a geometry column
+RENAME_COLUMNS = {
+    "boundary_condition_1d": [
+        ("boundary_type", "type"),
+    ],
+    "boundary_condition_2d": [
```
[the remark this answers is not in the document]
Comment (elisalle) [reply threedi_schema/migrations/versions/0225_migrate_lateral_boundary_condition_tables.py:85]: I just tried it out on an sqlite, and got this:
```
spatialite> pragma table_info(boundary_condition_2d);
0|id|INTEGER|1||0
1|display_name|VARCHAR(255)|0||0
2|timeseries|TEXT|0||0
3|type|INTEGER|0||0
4|geom|LINESTRING|1||0
5|code|TEXT|0||0
6|tags|TEXT|0||0
7|time_units|TEXT|0||0
8|interpolate|BOOLEAN|0||0
```
could you send me the sqlite you ran the migration on?

[the remark this answers is not in the document]
Comment (margrietpalm) [reply threedi_schema/migrations/versions/0225_migrate_lateral_boundary_condition_tables.py:85]: Hmm, cannot reproduce that one  any more.

File: threedi_schema/domain/models.py [hunk 2]
```diff
@@ -462,16 +479,20 @@ class TimeStepSettings(Base):
...
+    type = Column(IntegerEnum(constants.BoundaryType))
+    timeseries = Column(Text)
+    time_units = Column(Text)
+    interpolate = Column(Boolean)
+    geom = Column(Geometry("POINT"), nullable=False)
 
     connection_node_id = Column(
         Integer,
         ForeignKey(ConnectionNode.__tablename__ + ".id"),
-        nullable=False,
```
[the remark this answers is not in the document]
Comment (elisalle) [reply threedi_schema/domain/models.py:474]: Regarding the modelchecker check, wouldn't that already be covered by BoundaryCondition1DObjectNumberCheck?

[the remark this answers is not in the document]
Comment (margrietpalm) [reply threedi_schema/domain/models.py:474]: If that check indeed ensures that every `connection_node_id` that is used exists. However, I'm a bit confused by this test; I thought that the `connection_node_id` should match to a `v2_connection_nodes` entry.

[the remark this answers is not in the document]
Comment (elisalle) [reply threedi_schema/domain/models.py:474]: You're right, I misread the check.

[the remark this answers is not in the document]
Comment (elisalle) [reply threedi_schema/domain/models.py:474]: I've added the checks here https://github.com/nens/threedi-modelchecker/commit/c25c2e414aaf76cb8ecdaf07ee86efe5ee642d88

**Comment by elisalle** [comment]
> Also there are no changes added to `CHANGES.rst`.
the changelog has now been updated

**Review by margrietpalm — APPROVED** [review]

**Pull request merged**